keps/sig-storage/625-csi-migration/README.md

```diff
@@ -156,10 +163,209 @@ Running this set of tests is optional for a per-provider basis. We would
+
+
+### Monitoring Requirements
+
+* **How can an operator determine if the feature is in use by workloads?**
+  We will have metrics `csi_sidecar_duration_seconds` on the CSI sidecars and `storage_operation_duration_seconds` on the kube-controller-manager and kubelet side to indicate whether this operation is a migrated operation or not. These metrics will have a `migrated` field to indicate if this is a migrated operation.
+
+* **What are the SLIs (Service Level Indicators) an operator can use to determine
+the health of the service?**
+  - [x] Metrics
+    - Metric name: csi_sidecar_duration_seconds && storage_operation_duration_seconds, these metrics will have a `migrated` field
+    - [Optional] Aggregation method:
+    - Components exposing the metric: CSI sidecars, kubelet, kube-controller-manager
+  - [x] Other (treat as last resort)
+    - Details: Pod using in-tree plugin has failure.
+
+* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
+  - SLO with migration on matches the existing plugin's in-tree SLO with offset less than 1%
+
+* **Are there any missing metrics that would be useful to have to improve observability
+of this feature?**
+Node side CSI operation metrics. It will be implemented in the GA phase.
+
+### Dependencies
+
+
+* **Does this feature depend on any specific services running in the cluster?**
+
+  - Corresponding CSI Driver for in-tree CSI migration enabled plugin
+    - Usage description:
+      - Impact of its outage on the feature: in-tree plugin stops working without the CSI Driver properly setup
+      - Impact of its degraded performance or high-error rates on the feature: Error or performance decrease will be reflected for volume intensive operation
+
+
+### Scalability
+
+* **Will enabling / using this feature result in any new API calls?**
+  Yes.
+  - For volume attach/detach:
+    VolumeAttachment CREATE/PATCH APIs will be called for volume attachment by kube-controller-manager and csi-attacher. VolumeAttachment delete API will be called for volume detachment by csi-attacher. 1 for each needed.
+    VolumeAttachment LIST/WATCH api will be called by csi-attacher to monitor the VolumeAttachment.
+  - For volume provision/delete:
+    PVC LIST/WATCH apis will be called by csi-provisioner to monitor the PVC status. PV CREATE api will be called by csi-provisioner to create PV. PV DELETE api will be called by csi-provisioner to delete PV. PVC/PV PATCH api will be called by csi-provisioner for updating the object.
+    Notice that these new calls from csi-provisioner also mean that we will reduce call from the kube-controller-manager side.
+  - When CSI driver is being installed, there will be CSIDriver/CSINode CREATE/PATCH api call by kubelet. For each kubelet that installs the driver there will be one call.
```

Comment:
Note CSIDriver object is only one per driver and does not get regularly patched.

Reply:
Yes, for CSIDriver only one create call per driver. For CSINode, it is patch call per driver. Let me clarify the content